cmd/manager/main.go

```diff
@@ -176,11 +176,15 @@ func main() {
 		os.Exit(1)
 	}
 
-	certPool, err := httputil.NewCertPool(caCertDir)
+	certPool, err := httputil.NewCertPool(caCertDir, setupLog, func() {
+		setupLog.WithName("cert-pool").Info("restarting to reload CA certificates")
+		os.Exit(1)
```

Comment:
I'm a bit concerned with this approach from the perspective of a user that might see the pod having restarts and suspecting it is a bug, filing a bug report, and then us as maintainers having to perform deeper investigation to find that it was a cert rotation and stating that it is expected behavior for the pod to "crash" to reload CA certificates.

While it may be easier to simply crash the pod and it be restarted, I think it will throw a false negative signal to users. I think a caching based approach, while more complex, would ensure we aren't issuing false negative signals.

Reply:
It's clear from the (prior) logs that it's a restart from cert rotation.
CA certificates should rarely, if ever, rotate. We're talking a minimum of 3 months (for default cert-manager certificates) to years for commercial CAs.
According to Steve K, this is a common approach (I specifically asked him about it in another project).

Not sure what you mean by a cache-based approach? These are files mounted in the pod itself, not Secrets or ConfigMaps that are being monitored.